cmake/RAJAMacros.cmake

```diff
@@ -93,6 +93,48 @@ macro(raja_add_plugin_library)
 
 endmacro(raja_add_plugin_library)
 
+# Allows strings embedded in test files to used to process ctest results.
+# Only works for new testing framework/structure (no effect on old tests).
+# Borrowed heavily from CAMP.
+function(raja_set_failtest TESTNAME)
```

Comment:
Hi @trws @rhornung67 , this function is borrowed heavily from CAMP's version of dealing with fail tests. There are some modifications for RAJA and to be more conformant to BLT. Let me know if this is ok. Thanks!

Reply:
@MrBurmark has something working for HIP tests that are currently known failures. Can we apply the same technique for these cases?

Reply:
Talked with @MrBurmark, and his HIP tests are not trying to actively induce failures, which is what this function enables us to do.